### PR TITLE
Introspection guide should use the spk download script 

### PR DESCRIPTION
**Pull request by edaena**

### CODE REVIEW

**Review by mtarng — APPROVED** [review]

File: docs/service-introspection-onboarding.md [hunk 1]
```diff
@@ -125,10 +125,13 @@ Paste the following task in its corresponding `azure-pipelines.yml`:
     commitId=$(echo "${commitId:0:7}")
     service=$(Build.Repository.Name)
     service=${service##*/}
-    VERSION_TO_DOWNLOAD=$(curl -s "https://api.github.com/repos/CatalystCode/spk/releases/latest" | grep "tag_name" | sed -E 's/.*"([^"]+)".*/\1/')
-    echo "Downloading SPK version $VERSION_TO_DOWNLOAD" && wget "https://github.com/CatalystCode/spk/releases/download/$VERSION_TO_DOWNLOAD/spk-linux"
-    chmod +x ./spk-linux
-    ./spk-linux deployment create -n $(ACCOUNT_NAME) -k $(ACCOUNT_KEY) -t $(TABLE_NAME) -p $(PARTITION_KEY) --p1 $(Build.BuildId) --image-tag $tag_name --commit-id $commitId --service $service
+    echo "Downloading the latest SPK version"
```
Comment (samiyaakhtar) [inline docs/service-introspection-onboarding.md]: The download script in bedrock uses a variable to determine if a specific version is specified to use spk. Maybe the wording should say "Downloading SPK" to be less specific?

File: docs/service-introspection-onboarding.md [hunk 2]
```diff
@@ -161,11 +166,19 @@ Paste the following task towards the end of your release step in the release
...
 
 ```yaml
-latest_commit=$(git rev-parse --short HEAD)
-VERSION_TO_DOWNLOAD=$(curl -s "https://api.github.com/repos/CatalystCode/spk/releases/latest" | grep "tag_name" | sed -E 's/.*"([^"]+)".*/\1/') 
-echo "Downloading SPK version $VERSION_TO_DOWNLOAD" && wget "https://github.com/CatalystCode/spk/releases/download/$VERSION_TO_DOWNLOAD/spk-linux"
-chmod +x ./spk-linux
-./spk-linux deployment create  -n $(ACCOUNT_NAME) -k $(ACCOUNT_KEY) -t $(TABLE_NAME) -p $(PARTITION_KEY)  --p2 $(Release.ReleaseId) --hld-commit-id $latest_commit --env $(Release.EnvironmentName) --image-tag $(Build.BuildId)
+latest_commit=$(git rev-parse --short HEAD) VERSION_TO_DOWNLOAD=$(curl -s
+"https://api.github.com/repos/CatalystCode/spk/releases/latest" | grep
+"tag_name" | sed -E 's/.*"([^"]+)".*/\1/')
```
Comment (samiyaakhtar) [inline docs/service-introspection-onboarding.md]: Remove new line

File: docs/service-introspection-onboarding.md [hunk 3]
```diff
@@ -161,11 +166,19 @@ Paste the following task towards the end of your release step in the release
...
+echo "Downloading the latest SPK version"
+curl https://raw.githubusercontent.com/Microsoft/bedrock/master/gitops/azure-devops/build.sh > build.sh
+chmod +x build.sh
+. ./build.sh --source-only
+get_spk_version
+download_spk
+./spk/spk deployment create  -n $(ACCOUNT_NAME) -k
+$(ACCOUNT_KEY) -t $(TABLE_NAME) -p $(PARTITION_KEY)  --p2 $(Release.ReleaseId)
+--hld-commit-id $latest_commit --env $(Release.EnvironmentName) --image-tag
+$(Build.BuildId)
```
Comment (samiyaakhtar) [inline docs/service-introspection-onboarding.md]: These should be in one line

File: docs/service-introspection-onboarding.md [hunk 4]
```diff
@@ -179,14 +192,21 @@ that corresponds to the CD release pipeline.
...
+echo "Downloading the latest SPK version"
+curl https://raw.githubusercontent.com/Microsoft/bedrock/master/gitops/azure-devops/build.sh > build.sh
+chmod +x build.sh
+. ./build.sh --source-only
+get_spk_version
+download_spk
+./spk/spk deployment create  -n $(ACCOUNT_NAME) -k
+$(ACCOUNT_KEY) -t $(TABLE_NAME) -p $(PARTITION_KEY)  --p2 $(Build.BuildId)
+--hld-commit-id $latest_commit --env $(Build.SourceBranchName) --image-tag
+$tag_name
```
Comment (samiyaakhtar) [inline docs/service-introspection-onboarding.md]: same

File: docs/service-introspection-onboarding.md [hunk 5]
```diff
@@ -179,14 +192,21 @@ that corresponds to the CD release pipeline.
...
-latest_commit=$(git rev-parse --short HEAD)
-tag_name=$(Build.BuildId)
-VERSION_TO_DOWNLOAD=$(curl -s "https://api.github.com/repos/CatalystCode/spk/releases/latest" | grep "tag_name" | sed -E 's/.*"([^"]+)".*/\1/') 
-echo "Downloading SPK version $VERSION_TO_DOWNLOAD" && wget "https://github.com/CatalystCode/spk/releases/download/$VERSION_TO_DOWNLOAD/spk-linux"
-chmod +x ./spk-linux
-./spk-linux deployment create  -n $(ACCOUNT_NAME) -k $(ACCOUNT_KEY) -t $(TABLE_NAME) -p $(PARTITION_KEY)  --p2 $(Build.BuildId) --hld-commit-id $latest_commit --env $(Build.SourceBranchName) --image-tag $tag_name
+latest_commit=$(git rev-parse --short HEAD) tag_name=$(Build.BuildId)
+VERSION_TO_DOWNLOAD=$(curl -s
+"https://api.github.com/repos/CatalystCode/spk/releases/latest" | grep
+"tag_name" | sed -E 's/.*"([^"]+)".*/\1/')
```
Comment (samiyaakhtar) [inline docs/service-introspection-onboarding.md]: Same

File: docs/service-introspection-onboarding.md (hunk 1 above)
Comment (edaena) [reply docs/service-introspection-onboarding.md]: Are you suggesting I remove "latest SPK version" from that log text?

Comment (samiyaakhtar) [reply docs/service-introspection-onboarding.md]: Yup :)

**Review by samiyaakhtar — APPROVED** [review]

**Pull request merged**